src/openzaak/components/autorisaties/admin_views.py

```diff
@@ -236,17 +254,19 @@ def get_initial(applicatie: Applicatie) -> List[Dict[str, Any]]:
         grouped[key].append(autorisatie)
 
     # if there's no existing records yet, there will not be any autorisaties and we
-    # have to inject the autorisatiespec itself. See #1080 for the bug report.
-    for spec in autorisatie_specs.values():
-        key = _get_group_key(spec)
+    # have to inject the CatalogusAutorisatie itself. See #1080 for the bug report.
+    for catalogus_autorisatie in catalogus_autorisaties.values():
+        key = _get_group_key(catalogus_autorisatie[0])
```

Comment:
I'm not sure if this is possible without completely rewriting `get_initial_for_component` (and I'd rather not :grimacing:)

Reply:
I've split off a different function `get_initial_for_component_for_catalogus_autorisaties`, because `get_initial_for_component` was getting way too complicated and relied on workarounds to get the catalogus autorisaties is. Let me know what you think https://github.com/open-zaak/open-zaak/pull/1706/commits/1a49cd14842b6c5c93681fe4c3e7dc6498aaf8bd

Reply:
Nice refactoring! It's much more readable now!